pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp

```diff
@@ -517,40 +621,74 @@ class StateVectorKokkos final
         applyMatrix(matrix.data(), wires, inverse);
     }
 
+    /**
+     * @brief Apply a given matrix for controlled operations directly to the
+     * statevector using a raw matrix pointer vector.
+     *
+     * @param matrix Pointer to the array data (in row-major format).
+     * @param controlled_wires Controlled wires
+     * @param controlled_values Controlled values (true or false)
+     * @param wires Wires to apply gate to.
+     * @param inverse Indicate whether inverse should be taken.
+     */
+    inline void applyControlledMatrix(
```

Comment:
Where is this method tested with non-const `Complex *matrix`?

Reply:
test added for both applymatrix and applycontrolledmatrix with non-const complex *matrix